src/pipeline.py

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env python3
+
+from argparse import ArgumentParser, ArgumentTypeError
+from multiprocessing import Process, Pool, Queue
+from socket import gethostname
+
+class PipelineCLIParser(ArgumentParser):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.add_argument('--workers', type=positive_int_type, default=1,
+                help="Number of workers to process work", dest='n_workers')
+        self.add_argument('--mode', choices={'crawler', 'ucrawler', 'user'}, default='user',
+                help="Run mode. Either 'crawler', 'ucrawler', or 'user'")
+        self.add_argument('--source', required=True, help='HTTP link to firebase')
+        self.add_argument('--filter', nargs='+', help='A relational condition over metadata that we pull')
+
+def positive_int_type(val):
+    intval = int(val)
+    if intval <= 0:
+        raise ArgumentTypeError("%s is not a positive integer" % val)
+    return intval
+
+class PipelineWorker(Process):
+    def __init__(self, context, *args, **kwargs):
```

Comment:
Do we want to allow unnamed workers? If not, I think name should be an explicit argument instead of being passed in kwargs. Or have a check to ensure the name has been passed in.

Reply:
done